Rakefile

```diff
@@ -125,30 +137,39 @@ def replace_puppet_pins(passing_puppetagent_sha)
   File.write(BEAKER_OPTIONS_FILE, beaker_options_from_file.pretty_inspect)
 end
 
-namespace :puppet_submodule do
+namespace :dev_deps_update do
   desc 'update puppet submodule commit'
   task :update_puppet_version do
     #  ensure we fetch here, or the describe done later could be wrong
-    my_jenkins_passing_json = jenkins_passing_json_parsed
+    my_jenkins_passing_json = jenkins_passing_json_parsed(PUPPET_AGENT_BRANCH)
     git_checkout_command = "cd #{PUPPET_SUBMODULE_PATH} && git fetch origin && " \
       "git checkout #{lookup_passing_puppet_sha(my_jenkins_passing_json)}"
     puts("checking out known passing puppet version in submodule: `#{git_checkout_command}`")
     system(git_checkout_command)
     # replace puppet-agent sha pin in beaker options file
     replace_puppet_pins(lookup_passing_puppetagent_sha(my_jenkins_passing_json))
   end
+  desc 'update facter submodule commit'
+  task :update_facter_version do
```

Comment:
might be worth putting this logic into a separate function, since it's nearly identical to the task for puppet?